server/routes/probationPractitionerReferrals/probationPractitionerReferralsController.ts

```diff
@@ -174,10 +174,27 @@ export default class ProbationPractitionerReferralsController {
       ActionPlanUtils.getLatestApprovedActionPlanSummary(approvedActionPlanSummaries)
     let actionPlanAppointments: ActionPlanAppointment[] = []
     if (latestApprovedActionPlanSummary !== null) {
-      actionPlanAppointments = await this.interventionsService.getActionPlanAppointments(
-        accessToken,
-        latestApprovedActionPlanSummary.id
-      )
+      actionPlanAppointments = await this.interventionsService
+        .getActionPlanAppointments(accessToken, latestApprovedActionPlanSummary.id)
+        .then(actionPlanAppointmentsReturned => {
+          return actionPlanAppointmentsReturned
+            .flatMap(x => {
+              return [
+                x,
+                x.oldAppointments
+                  ? x.oldAppointments!.map(y => {
+                      const actionPlanAppointment: ActionPlanAppointment = {
+                        sessionNumber: x.sessionNumber,
+                        ...y,
+                      }
+                      return actionPlanAppointment
+                    })
+                  : [],
+              ]
+            })
+            .flat()
+            .filter(x => x)
```

Comment:
this isn't going to be clear to anyone looking at this code in the future, either add an explanation of why you are using flat and filter (particularly filter x => x which isn't immediately why you would need this) or change to use the spread operator as previously recommended (which removes the need to iterate through the array again to perform flat and filter)